### PR TITLE
VB-3054 Update non-association event endpoint and data processing - s…

### PR DESCRIPTION
**Pull request by AledWEvans**
VB-3054 Update non-association event endpoint and data processing - stage one removal of effectiveDate and expiryDate now uses new api

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/client/NonAssociationsApiClient.kt [hunk 1]
```diff
@@ -19,22 +19,23 @@ class NonAssociationsApiClient(
...
   }
 
-  fun getOffenderNonAssociation(offenderNo: String): OffenderNonAssociationDetailsDto? {
-    LOG.debug("Entered getOffenderNonAssociation $offenderNo")
+  fun getPrisonerNonAssociation(prisonerNumber: String): PrisonerNonAssociationDetailsDto? {
+    LOG.debug("Entered getPrisonerNonAssociation $prisonerNumber")
+
     return webClient.get()
-      .uri("/legacy/api/offenders/$offenderNo/non-association-details?currentPrisonOnly=true&excludeInactive=true")
+      .uri("/prisoner/$prisonerNumber/non-associations?includeOtherPrisons=true")
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/client/NonAssociationsApiClient.kt:29]: should includeOtherPrisons be false or will it not matter as we do a prison check later?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt [hunk 2]
```diff
@@ -468,7 +468,7 @@ class SessionServiceTest {
     }
 
     @Test
-    fun `session does not contain conflicts when an offender has no non-associations and no double bookings`() {
+    fun `session does not contain conflicts when an prisoner has no non-associations and no double bookings`() {
```
Comment (dhirajshettymoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt:471]: very minor typo error - an prisoner should be a prisoner wherever applicable

**Review by dhirajshettymoj — APPROVED** [review]
LGTM - 2 minor comments to look at.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/client/NonAssociationsApiClient.kt (hunk 1 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/client/NonAssociationsApiClient.kt:29]: includeOtherPrisons not sure really,... could be false,.. just was a little concerned if a prisoner had been transferred temporarily but the visit is later on,...

**Pull request merged**